bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/scheduling/Strand.java

```diff
@@ -71,15 +70,14 @@
     public boolean blockedOnExtern;
     public Set<ChannelDetails> channelDetails;
     public Set<SchedulerItem> dependants;
-    public ObserverContext observerContext;
     public boolean cancel;
 
     SchedulerItem schedulerItem;
     List<WaitContext> waitingContexts;
     WaitContext waitContext;
     ItemGroup strandGroup;
 
-    private Map<String, Object> globalProps;
+    public Map<String, Object> globalProps;
     public TransactionLocalContext currentTrxContext;
```

Comment:
Can we move the transaction context to strand locals similar to observation context

Reply:
yes, we need to do that. But will be a separate pr because it's tightly coupled to transaction langlib.